sympy/physics/units/__init__.py

```diff
@@ -222,23 +222,28 @@ def find_unit(quantity):
 
```

Comment:
I think docstring should explain its exact behavior

Reply:
@ylemkimon, I have made the requested changes, please have a look.

Reply:
I think `all instances of units/dimensions with these arguments.` is little ambiguous and it doen.
If a string is given, it currently returns
(i) all units matching the string
(ii) if a dimension exactly matching the string exists, all units with that dimension
(iii) if a unit exactly matching the string exists, all units with same dimension